Static head

IngelaLind opened the pull request:
Added components:
Boundaries.AccelerationBoundary; sets a global acceleration vector
Processes.StaticHead; adds or subtracts static head depending on acceleration and positions
Boundaries.TankCuboid; a tank for incompressible + gas media, which takes into account the global acceleration and the positions of an arbitrary number of inlets and outlets. This one has cuboid geometry and disregards acceleration in y-direction.
Boundaries.Internal.PartialTank; all the media related equations for a tank, setup to make it easy to model an arbitrary geometry by extension of this model
Boundaries.Tests; eight different test models for tanks and staticHead
Media.myMedia.Incompressible.Examples.Dowcal100; from public data
Media.myMedia.Incompressible.Examples.TestDowcal100
Media.myMedia.Incompressible.Examples.TestLiqLoopDowcal100
Media.myMedia.Incompressible.Examples.Dowcal100E; from public data
Media.myMedia.Incompressible.Examples.TestDowcal100E
Media.myMedia.Incompressible.Examples.JP8; Jet propulsion fuel
Media.myMedia.Incompressible.Examples.TestJP8
Media.myMedia.Incompressible.Examples.TestLiqLoopJP8
Media.myMedia.GasAndIncompressible.PartialGasAndIncompressible; base class to be able to combine e.g. fuel and air
Media.myMedia.GasAndIncompressible.JP8DryAir
Media.myMedia.GasAndIncompressible.Examples; four test models for JP8DryAir

dzimmer commented:
First of all: Many thanks! That is quite a contribution. You really seem to have solved the general case.

- The main design choices meaning which components to add to the library seem correct to me. This all makes perfect sense.
- The value of having a medium model that handles gas and liquid state really shows here. It would be great to have water / air as a second implementation. 
- Minor cosmetics needed for the global acceleration model. setFromExpression serves no real purpose and can be removed. Icon needs a little improvement
- without further understanding I would prefer to remove the chaoticLife functionality at least for now. (may still be added later)

Do not take this comment as cause for any action. It primarily serves documenting my state of analysis. Certain modifications are also easier to do on our side.
Code analysis continues....

dzimmer commented:
Further small issues:
- The code contains various fudge constants which should be turned into actual parameters / constants
- Some more documentation  and explanation needs to be added. I think, I understand the code now... 
- The authorship must be stated in the revision element of the models

I will do these changes and push to this pull-request repo branch.. Ingela, please make sure that I or contributors have rights to push on this branch. Thx

dzimmer commented:
Ingela, I will proceed as detailed in:
https://gist.github.com/wtbarnes/56b942641d314522094d312bbaf33a81

but not today anymore...

IngelaLind commented:
Dear Dirk,
I’m happy you liked my contribution.  I have no reservations on the changes you suggest. Do I understand you right that a fudge constant is something declared as Real x=2; without a constant or parameter declaration in front?
I am sure that more explanations are needed! It is just so hard to tell which ones, when those I work with have no trouble understanding the new parts I added, which they are familiar with from previous models, but still struggles with the basic concept of the library.
I totally forgot about authorship. Thanks for pointing that out.

It should be very easy to add air/water as a second GasAndIncompressible media model. Do you want me to do that?

I have been tinkering with the icon for the acceleration component the last weeks (the icon layer is REALLY hard to work on!) to add a dynamic indicator of the direction. This is a reasonable request from my colleagues. The directions of the acceleration is actually one of the things that is easiest to get wrong in day-to-day work. Do you want that now as well?

I have also added our logotype to this mail, if you want to add it to your contributors page.
Best regards,
Ingela

Ingela Lind, M Sc, Ph D
Technical Fellow
Simulation and Thermal Analysis
Vehicle Systems
SAAB Aerosystems

***@***.***
From: Dirk Zimmer ***@***.***>
Sent: den 7 juni 2024 19:06
To: DLR-SR/ThermofluidStream ***@***.***>
Cc: Lind Ingela ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [DLR-SR/ThermofluidStream] Static head (PR #182)


Further small issues:

  *   The code contains various fudge constants which should be turned into actual parameters / constants
  *   Some more documentation and explanation needs to be added. I think, I understand the code now...
  *   The authorship must be stated in the revision element of the models

I will do these changes and push to this pull-request repo branch.. Ingela, please make sure that I or contributors have rights to push on this branch. Thx

—
Reply to this email directly, view it on GitHub<https://smex-ctp.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fgithub.com%2fDLR%2dSR%2fThermofluidStream%2fpull%2f182%23issuecomment%2d2155214694&umid=8a9e28c7-7782-4043-856e-7d9e5ec94d70&auth=ff0b6f163eda6bf665d54d621d177fa8dadd9ca8-a9923b0766a3edad775396b22d1cdd27511161c4>, or unsubscribe<https://smex-ctp.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fgithub.com%2fnotifications%2funsubscribe%2dauth%2fBGLCL7653ACTIK7JAAAW7DDZGHR7ZAVCNFSM6AAAAABHDF3AC6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNJVGIYTINRZGQ&umid=8a9e28c7-7782-4043-856e-7d9e5ec94d70&auth=ff0b6f163eda6bf665d54d621d177fa8dadd9ca8-a4cc1535d54a4601cd162a403e55e0a575d272ee>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

dzimmer commented:
work to be continued. finished for today

IngelaLind commented:
It is not as easy as I thought to add a water/air media since the incompressible media has no common base class with any of the water models. I had expected at least partialMedium in common. I do not know at the moment if it is easier to add another simplified water in the incompressible base class or do the better thing and make a GasAndCompressible partial class or rework the GasAndIncompressible to be more flexible. I do not have any motivation other than my own curiosity to do any of it and since we have no interest in water tanks at the moment,  it is hard to get that work funded.

dzimmer commented:
Thanks for the warning. Our application would be to model a water extractor (and a corresponding water injector) A colleague of mine could attempt it but he is less familiar with Media modelling

IngelaLind commented:
To do a good water extractor you would need to work with a media with the components dry air +water vapour + free water. That is more complex than the dry air + fuel mixture I did, where the fuel vapour is totally neglected. It is a very good idea to do one, though. We would certainly need it for our applications.

From: Dirk Zimmer ***@***.***>
Sent: den 13 juni 2024 20:06
To: DLR-SR/ThermofluidStream ***@***.***>
Cc: Lind Ingela ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [DLR-SR/ThermofluidStream] Static head (PR #182)


Thanks for the warning. Our application would be to model a water extractor (and a corresponding water injector) A colleague of mine could attempt it but he is less familiar with Media modelling

—
Reply to this email directly, view it on GitHub<https://smex-ctp.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fgithub.com%2fDLR%2dSR%2fThermofluidStream%2fpull%2f182%23issuecomment%2d2166467404&umid=c6945c36-048b-4c5b-827e-f8d80a7bc3f5&auth=470be8387acc995b722b438d7923feea0fe8a5b3-9aaa719810c7e4bea29996f24d8f9d9e8c686381>, or unsubscribe<https://smex-ctp.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fgithub.com%2fnotifications%2funsubscribe%2dauth%2fBGLCL7Y24BVQ5UPPSGFYDD3ZHHNQRAVCNFSM6AAAAABHDF3AC6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNRWGQ3DONBQGQ&umid=c6945c36-048b-4c5b-827e-f8d80a7bc3f5&auth=470be8387acc995b722b438d7923feea0fe8a5b3-3c1e699ea2f9af44528615c30cf805f58fdaeb54>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

dzimmer commented:
Alright. I have added a color scheme. We have used light green for all those components that use dynamic pressure. 
Now I have introduced brown/orange to mark components using the gravity field.
It remains an open design question whether to include the gravity field into the drops of common but (also to avoid conflicts with parallel developments), we will not do that now but may do so in a next version. Hence the warning in the doc of the components

dzimmer commented:
Ingela, Can you check the simulation results on your side? My modifications should not have changed anything.
If you give thumbs up. I'll accept the Pull request

dzimmer commented:
> To do a good water extractor you would need to work with a media with the components dry air +water vapour + free water. That is more complex than the dry air + fuel mixture I did, where the fuel vapour is totally neglected. It is a very good idea to do one, though. We would certainly need it for our applications.

A colleague of mine has implemented a water extractor volume using water and moist air as seperate media. However this has its complications when the tanks drains. I propose that you do both a common effort.

IngelaLind commented:
Good. Then I should inform you that the present version of the acceleration boundary is simplified and not complete. For a large aircraft or other vessels moving in 3D, the rotation and center of rotation would also be needed. I do have the necessary equations somewhere, but it was not a priority now. So, the current version is about a third of a complete set of variables.

From: Dirk Zimmer ***@***.***>
Sent: den 14 juni 2024 17:53
To: DLR-SR/ThermofluidStream ***@***.***>
Cc: Lind Ingela ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [DLR-SR/ThermofluidStream] Static head (PR #182)


Alright. I have added a color scheme. We have used light green for all those components that use dynamic pressure.
Now I have introduced brown/orange to mark components using the gravity field.
It remains an open design question whether to include the gravity field into the drops of common but (also to avoid conflicts with parallel developments), we will not do that now but may do so in a next version. Hence the warning in the doc of the components

—
Reply to this email directly, view it on GitHub<https://smex-ctp.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fgithub.com%2fDLR%2dSR%2fThermofluidStream%2fpull%2f182%23issuecomment%2d2168314093&umid=d2d10bd8-ee4d-4b0a-8faa-ab2f4706033c&auth=a8e86023f0944e40d564f17efe6f92d5b33adf5f-d49d7385e7ea3a39c0e4a6eaa3a5d54420528f34>, or unsubscribe<https://smex-ctp.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fgithub.com%2fnotifications%2funsubscribe%2dauth%2fBGLCL7YVYCTOGD5QZNOAXETZHMGW7AVCNFSM6AAAAABHDF3AC6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNRYGMYTIMBZGM&umid=d2d10bd8-ee4d-4b0a-8faa-ab2f4706033c&auth=a8e86023f0944e40d564f17efe6f92d5b33adf5f-5bc7025a5aafd4cbbc45de94053df311dd59108c>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

IngelaLind commented:
Of course! But after an entire morning spent on git/gitHub trouble and reconfiguration,  I have run out of time. I will do the testing earliest next Monday.
---------

Ingela, Can you check the simulation results on your side? My modifications should not have changed anything.

tobolar commented on ThermofluidStream/Boundaries/Tests/DynamicBoundaries.mo:
Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

tobolar commented on ThermofluidStream/Boundaries/Tests/PhaseSeperator.mo:
Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

tobolar commented on ThermofluidStream/Boundaries/Tests/Reservoir.mo:
Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

tobolar commented on ThermofluidStream/Boundaries/Tests/Volumes.mo:
Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

tobolar commented on ThermofluidStream/Processes/FlowResistance.mo:
Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

tobolar commented on ThermofluidStream/Processes/Tests/ConvectivePipe.mo:
Changes like this do not contribute to the topic of this PR and shall be ommited by commit.

tobolar commented on ThermofluidStream/Processes/Tests/ConvectivePipe_serial.mo:
Changes like this do not contribute to the topic of this PR and shall be ommited by commit.
